pynest/nest/lib/hl_api_types.py

```diff
@@ -1286,6 +1286,10 @@ def serialize_data(data):
 
     if isinstance(data, (numpy.ndarray, NodeCollection)):
         return data.tolist()
+    if isinstance(data, (numpy.int_, numpy.intc, numpy.intp, numpy.int8,
+                            numpy.int16, numpy.int32, numpy.int64, numpy.uint8,
+                            numpy.uint16, numpy.uint32, numpy.uint64)):
+        return int(data)
```

Comment:
You could maybe make use of `np.integer`, instead of explicitly listing most of the integer types.
```suggestion
    if np.issubdtype(d, np.integer):
        return int(data)
```

Reply:
Yes, indeed, it works as well, thanks!